src/Modules/Scoreboard/Services/ScoreboardService.cs

```diff
@@ -0,0 +1,144 @@
+﻿using EvoSC.Common.Config.Models;
+using EvoSC.Common.Interfaces;
+using EvoSC.Common.Interfaces.Services;
+using EvoSC.Common.Services.Attributes;
+using EvoSC.Common.Services.Models;
+using EvoSC.Common.Util;
+using EvoSC.Manialinks.Interfaces;
+using EvoSC.Modules.Official.Scoreboard.Interfaces;
+
+namespace EvoSC.Modules.Official.Scoreboard.Services;
+
+[Service(LifeStyle = ServiceLifeStyle.Singleton)]
+public class ScoreboardService : IScoreboardService
+{
+    private readonly IManialinkManager _manialinks;
+    private readonly IPlayerManagerService _playerManager;
+    private readonly IServerClient _server;
+    private readonly IEvoScBaseConfig _config;
+
+    private int _roundsPerMap = -1;
```

Comment:
Because these two variables are mutable within a multithreaded environment, I recommend locking them to avoid race conditions. This is something you should always consider when creating a Singleton service. More info on locking in C#: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/lock

Remember to lock on a separate object than the variables themselves to prevent deadlocks.


For example:
```csharp
private int _roundsPerMap = -1;
private readonly object _roundsPerMapLock = new();

// ...

lock(_roundsPerMapLock) {
   // access _roundsPerMap here ...
}
```